### ISSUE TITLE
Passwordless authentication no triggeres despite being the first configured alternative option

### ISSUE THREAD

**Issue by codependent** [keycloak/keycloak#13248]
### Describe the bug

The issue is thoroughly described, with the necessary configuration and screenshots, in the following Stackoverflow question: https://stackoverflow.com/questions/73034293/keycloak-doesnt-offer-passwordless-authentication-as-the-first-option-during-lo

Summing up: passwordless authentication has been configured as the first alternative auth mechanism. However Keycloak offers first the second alternative (password auth) to the user instead of the passworldless one.

### Version

18.0.2

### Expected behavior

The user should be prompted with the Sign in with Security Key option first

### Actual behavior

The user is asked for their password

### How to Reproduce?

Follow the configuration shown in https://stackoverflow.com/questions/73034293/keycloak-doesnt-offer-passwordless-authentication-as-the-first-option-during-lo

**Comment by codependent** [comment]
I've also verified that this behaviour happens on a Loginess+Passwordless scenario:

![image](https://user-images.githubusercontent.com/6399869/181008384-1beeac9a-5d4e-4601-a026-3edcecf7f68c.png)
With this configuration the user should be asked first for its loginless credentials but instead gets the user+password form:

![image](https://user-images.githubusercontent.com/6399869/181008523-e40287b1-3bb6-48e1-b20d-d566340bd52f.png)

So I have to "Try Another Way" to access the Sign in with security key screen:

![image](https://user-images.githubusercontent.com/6399869/181008653-3e614094-ec8e-4157-9750-4481cd12ec90.png)

**Comment by sventorben** [comment]
I am seeing a similar behaviour with a custom authenticator. It looks like this is caused by the logic implemented in [AuthenticationSelectionResolver](https://github.com/keycloak/keycloak/blob/18.0.2/services/src/main/java/org/keycloak/authentication/AuthenticationSelectionResolver.java#L63). From the code it looks like this is intended. However, I find this very confusing, because there is an option to sort alternative authenticators and I think they should be executed by given priority.

**Comment by sventorben** [comment]
@codependent Could you please check if the order/position of the credentials does affect the presented alternative?

![image](https://user-images.githubusercontent.com/12183470/181104758-aead59d2-ee88-4504-95b7-8ed2dc806a47.png)

**Comment by codependent** [comment]
@sventorben I can't reorder the credentials, this is how they appear in the user credential management screen:

<img width="1491" alt="image" src="https://user-images.githubusercontent.com/6399869/181110196-2d805339-d529-4d06-a1b2-11352d8cb2eb.png">

**Comment by sventorben** [comment]
Your users are stored in LDAP/AD, right? Then the password isn't stored in Keycloak and seems to have preference. I am not sure if this is expected behavior, though.

**Comment by codependent** [comment]
@sventorben I'll have a look at the source code, thanks for pointing at it.

In any case it seems kind of misleading and I think it should follow the priorities defined in the Authentication Flow definition, shouldn't it?

In fact, [according to the documentation](https://www.keycloak.org/docs/latest/server_admin/index.html#_webauthn_passwordless), it sounds as if once the user has set up passwordless auth it shouldn't be prompted for the password again:

> [...] During the first authentication, the user must use the password and second-factor WebAuthn credential. **The user does not need to provide the password and second-factor WebAuthn credential if they use the WebAuthn Passwordless credential.**

**Comment by sventorben** [comment]
> it should follow the priorities defined in the Authentication Flow

Agree. Imho using the priorities of credentials only makes sense if there are multiple of the same credential type, like multiple registered OTP devices.

**Comment by mabartos** [comment]
Related issue: #12102

**Comment by ririsoft** [comment]
> @codependent Could you please check if the order/position of the credentials does affect the presented alternative?
> 
> ![image](https://user-images.githubusercontent.com/12183470/181104758-aead59d2-ee88-4504-95b7-8ed2dc806a47.png)

I confirm order/position of the user credentials set in the admin console works.
IMHO this bug is a duplicate of #12102

**Comment by mposolda** [comment]
Duplicates #12102

I am closing as a duplicate. Please continue in the other issue for the further discussions/questions/suggestions around this.
